component/prometheus/scrape/scrape.go

```diff
@@ -99,7 +99,12 @@ func (arg *Arguments) UnmarshalRiver(f func(interface{}) error) error {
 	*arg = DefaultArguments
 
 	type args Arguments
-	return f((*args)(arg))
+	err := f((*args)(arg))
+	if err != nil {
+		return err
+	}
+
+	return arg.HTTPClientConfig.Validate()
```

Comment:
Can you call out in a comment here that squashed blocks don't have UnmarshalRiver called, so we need to manually call the validate method?

Reply:
all set